File: src/app/interfaces/IShape.ts
import { Point } from "../classes/Point";
import { Color } from "../classes/Style";
import { IDimension } from "./IDimension";

export interface IShape {

    getId(): number;
    getType(): string;
    draw(p: Point, dimensions: Array<number>): void;
    move(newCenter: Point): void;
    resize(location: string, offset: Point, mouse: Point): void;
    copy(): IShape;
    delete(): void;
    setFill(color: Color): void;
    setStroke(color: Color): void;
}

File: src/main/java/com/PaintApp/PaintAppBackend/Interfaces/IShape.java
package com.PaintApp.PaintAppBackend.Interfaces;
import com.PaintApp.PaintAppBackend.Shapes.*;

public interface IShape {
    public int getId();
    public String getType();
    public void draw(Point p);
    public void move(Point newCenter);
    public void resize(Point newCenter, int[] newDimensions);
    public IShape copy();
    public void setFill(Color color);
    public void setStroke(Color color);
}
